### DISCUSSION TITLE
why leading and tracking instead of line-height and letter-spacing?

### DISCUSSION THREAD

**Discussion by codeflorist** [tailwindlabs/tailwindcss#9278] [Ideas]
for a utility css framework, the obvious class-names for me would be the names of the actual css-directives.

what i find myself doing over and over is forgetting, that in tailwind `line-height` is called `leading` and `letter-spacing` is called `tracking`.

both expressions...

- are very unintuitive
- can easily be cofnused
- are named completely different from their css directives
- afaik have no other occurance in web-development whatsoever (never came across them in 20 years of webdeving)

and each time i come across this i look to the heavens and shout `WHY? OH WHY?`.

does anyone have an answer for me?

**Comment by wongjn** [comment]
I believe for the origin of the terms:

`leading` for `line-height` comes from its meaning, "the distance between a pair of adjacent lines in composed text" in typography[^1]
`tracking` for `letter-spacing` comes from its meaning, "the standard distance between adjacent characters (as in composed text)"[^2]

They may have been picked because they are shorter than the CSS property counterparts.

[^1]: https://www.merriam-webster.com/dictionary/leading  
[^2]: https://www.merriam-webster.com/dictionary/tracking

**Comment by jenstornell** [comment]
I've seen updates from two word names to a single word names here and there. Two words are definitely the reason for this change. Personally I would prefer something like this, I think:

```
lh-1
lh-1.25
lh-1.5
lh-1.75
```

The same goes for `ls-`. Probably a bit abstract but to me leading and tracking are quite abstract as well.

**Comment by adamwathan** [comment]
Can see a bunch of old discussions about this from years ago:

https://github.com/tailwindlabs/tailwindcss/pull/664
https://github.com/tailwindlabs/tailwindcss/issues/667

Originally the names were chosen because they felt like good names at the time when the framework had a smaller surface area and we were trying to be expressive with the names vs. being purely mechanical and just mimicking the CSS properties. We had lots of stuff like this in the 0.x days, like `top-0` was actually just `pin-t` for example. Considered changing them but felt like it wasn't worth the pain of a breaking change.

**Reply by jenstornell** [reply]
Interesting to hear the history behind it. I've changed my answer just a bit, because I've always liked numbers better. Easier for me to remember and allows for a bigger scale.

About breaking changes. It's probably nice to not break sites with new updates. However, if we don't have breaking changes, things will not improve. Did you ever regret making a breaking change? If not, you should probably break stuff from time to time.
